Improve Lua support

razvand opened the pull request:
Add template configuration files and draft instructions for Lua support in Unikraft.

razvand committed:
Move helloworld.lua in fs0/

`fs0/` stores the filesystem information. This is where the Lua test
script (`helloworld.lua`) is now located.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

razvand committed:
Add .gitignore

Add `.gitgnore` file to ignore output files and folders.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

razvand committed:
kraft.yaml: Add effective targets

Add targets for KraftKit in `kraft.yaml` file. Each target identifies
the platform, VMM, architecture and the filesystem support (9pfs or
initrd).

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

razvand committed:
Add .config.* template files

These template configuration files are to be used by the `make
defconfig` command to generate the actual `.config` file. The full
command will be:

```
UK_DEFCONFIG=$(pwd)/.config.... make defconfig
```

This will generate the `.config` file in a non-interactive mode,
providing basis for automation.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

RaduNichita commented on README.md:
````diff
@@ -1,43 +1,66 @@
-Configure the application:
-```
-kraft configure
+```console
````
```suggestion
## Configure 

```console
```

RaduNichita commented on README.md:
````diff
@@ -1,43 +1,66 @@
-To configure it with the desired features:
-```
-make menuconfig
+```console
````
I think it's better to have some scripts to run the app, similar to [here](https://github.com/unikraft/app-redis)

RaduNichita commented on README.md:
````diff
@@ -1,43 +1,66 @@
 ```
 
-Run the application:
+```console
````
I think you should say what is the purpose of this command

RaduNichita commented on README.md:
````diff
@@ -1,43 +1,66 @@
-kraft run -i helloworld.lua
+```console
+make distclean
+UK_DEFCONFIG=$(pws)/.config... make defconfig
````
I think here we should give an example for the .... Otherwise, I would be tempted to run the command like it is.

RaduNichita commented on README.md:
```diff
@@ -1,43 +1,66 @@
 # Lua on Unikraft
```
Provide a short description on what is the purpose of the application, similar to [here](https://github.com/unikraft/app-nginx)

RaduNichita commented on README.md:
```diff
@@ -1,43 +1,66 @@
+kraft build -t lua-qemu-arm64-9pfs -j $(nproc)
+kraft build -t lua-qemu-arm64-initrd -j $(nproc)
+kraft build -t lua-fc-x86_64-initrd -j $(nproc)
+kraft build -t lua-fc-arm64-initrd -j $(nproc)
```
I have one question: is there a particular reason for which the running of the application is done without `kraft run`?

RaduNichita reviewed (changes requested):
Thanks, @razvand. 

I left some comments, I will test tommorow the application.

RaduNichita commented:
Should we have add a config for `arm64` for Firecracker, @razvand?

StefanJum reviewed (commented):
The commands work fine, all good besides the README file that needs updating to match the other applications.

razvand commented:
> Should we have add a config for `arm64` for Firecracker, @razvand?

I wasn't able to get that to run, so not yet. But soon.

StefanJum commented on README.md:
````diff
@@ -1,43 +1,463 @@
+Use the `.config.lua-qemu-aarch64-9pfs` configuration file together with `make defconfig` to create the configuration file:
+
+```console
+UK_DEFCONFIG=$(pwd)/.config.lua0qemu-aarch64-9pfs make defconfig
````
```suggestion
UK_DEFCONFIG=$(pwd)/.config.lua_qemu-aarch64-9pfs make defconfig
```

StefanJum reviewed (changes requested):
Almost there. The last 2 commits show up as `razvand authored and jobpaardekooper committed`, if you took the commits from some other branch or pr please add Job's `signed-off`/`co-authored` tag.

razvand committed:
README.md: Update README with new build & run instructions

Add detailed instructions on using KraftKit and the basic Make-based
system to configure, build and run Lua as a Unikraft application.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

razvand committed:
Add Firecracker configuration for x86_64

Firecraker requires a JSON file with the options to run a virtual
machine. This commit adds that file for x86_64.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

RaduNichita commented on README.md:
````diff
@@ -1,43 +1,463 @@
 ```
-kraft build
+
+Otherwise, we recommend building for `qemu/x86_64` for an `initrd` support filesystem like so:
````
```suggestion
Otherwise, we recommend building for `qemu/x86_64` with `initrd` support filesystem like so:
```

RaduNichita reviewed (changes requested):
Looks very good, thanks @razvand. One small comment from me.

StefanJum reviewed (approved):
All good on my side, @RaduNichita please review it and I'll add the tag after you're ok

RaduNichita reviewed (approved):
All good now. Many thanks @razvand 

Reviewed-by: Radu Nichita <radunichita99@gmail.com>

StefanJum reviewed (approved):
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum reviewed (approved):
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

RaduNichita reviewed (approved):
All good.

Reviewed-by: Radu Nichita <radunichita99@gmail.com>
